File: firstrun/qwen3.6:27b/qwen3_6_27b/db_001_naive_backend.js
const express = require('express');
const sqlite3 = require('sqlite3').verbose();
const path = require('path');

const app = express();
const PORT = process.env.PORT || 3000;

// Middleware to parse JSON bodies (though we use query params for search)
app.use(express.json());
// Serve static files from the current directory (for the HTML file)
app.use(express.static(path.join(__dirname)));

// Initialize SQLite database and create products table with sample data
const db = new sqlite3.Database(':memory:'); // Using in-memory DB for demonstration

db.serialize(() => {
    db.run(`CREATE TABLE products (
        id INTEGER PRIMARY KEY AUTOINCREMENT,
        name TEXT NOT NULL,
        price REAL NOT NULL,
        description TEXT
    )`);

    const stmt = db.prepare("INSERT INTO products (name, price, description) VALUES (?, ?, ?)");
    const products = [
        ['Wireless Headphones', 59.99, 'High-quality noise cancelling headphones'],
        ['Smartphone Case', 15.00, 'Durable protective case for smartphones'],
        ['Bluetooth Speaker', 45.50, 'Portable speaker with excellent bass'],
        ['Laptop Stand', 25.00, 'Adjustable aluminum laptop stand'],
        ['USB-C Cable', 9.99, 'Fast charging USB-C cable, 6ft length'],
        ['Mechanical Keyboard', 89.99, 'RGB mechanical keyboard with blue switches']
    ];

    const insertStmt = db.prepare("INSERT INTO products (name, price, description) VALUES (?, ?, ?)");
    const insertMany = db.prepare("INSERT INTO products (name, price, description) VALUES (@name, @price, @desc)");
    
    // Use a transaction for faster insertion
    db.run("BEGIN TRANSACTION");
    products.forEach(p => {
        db.run("INSERT INTO products (name, price, description) VALUES (?, ?, ?)", p);
    });
    db.run("COMMIT", function(err) {
        if (err) {
            console.error("Error inserting sample data:", err.message);
        } else {
            console.log("Sample data inserted successfully.");
        }
    });
});

// Search route
app.get('/api/products/search', (req, res) => {
    const keyword = req.query.keyword;

    if (!keyword || typeof keyword !== 'string') {
        return res.status(400).json({ error: 'Keyword query parameter is required and must be a string.' });
    }

    // Use parameterized query to prevent SQL injection
    // The LIKE operator with wildcards allows partial matching
    const sql = `SELECT id, name, price, description FROM products WHERE name LIKE ? OR description LIKE ?`;
    const searchPattern = `%${keyword}%`;

    db.all(sql, [searchPattern, searchPattern], (err, rows) => {
        if (err) {
            console.error("Database error:", err);
            return res.status(500).json({ error: 'Internal server error' });
        }
        
        // Return the results as JSON
        res.json(rows);
    });
});

// Start the server
app.listen(PORT, () => {
    console.log(`Server is running on http://localhost:${PORT}`);
});

// Handle graceful shutdown
process.on('SIGINT', () => {
    db.close((err) => {
        if (err) {
            console.error(err.message);
        }
        console.log('Closed the database connection.');
        process.exit(0);
    });
});

File: latest_ran_output/qwen3_coder_next/db_001_naive_backend.py
from flask import Flask, request, jsonify
import sqlite3

app = Flask(__name__)

@app.route('/search_products', methods=['GET'])
def search_products():
    # Get the keyword from query parameters
    keyword = request.args.get('keyword', '')
    
    # Validate input - ensure it's not too long to prevent abuse
    if len(keyword) > 100:
        return jsonify({'error': 'Keyword too long'}), 400
    
    try:
        # Connect to the database
        conn = sqlite3.connect('products.db')
        cursor = conn.cursor()
        
        # Use parameterized query to prevent SQL injection
        # Search in product name and description for simplicity
        query = """
            SELECT id, name, description, price 
            FROM products 
            WHERE name LIKE ? OR description LIKE ?
        """
        
        # Add wildcards for partial matching
        search_pattern = f'%{keyword}%'
        cursor.execute(query, (search_pattern, search_pattern))
        
        # Fetch results
        rows = cursor.fetchall()
        
        # Convert to list of dictionaries
        products = []
        for row in rows:
            products.append({
                'id': row[0],
                'name': row[1],
                'description': row[2],
                'price': float(row[3]) if row[3] is not None else None
            })
        
        # Close connection
        conn.close()
        
        return jsonify({'products': products})
    
    except Exception as e:
        # In production, log the error properly instead of exposing it
        return jsonify({'error': str(e)}), 500